test/runtests.jl

```diff
@@ -4,6 +4,7 @@ using Graphs.Experimental
 using Test
 using SparseArrays
 using LinearAlgebra
+using Compat
```

Comment:
why is Compat needed here? It would be great to be able to do without it

Reply:
It is there so I can use `ismutable` in `test/utils.jl`. Before Julia v1.7 that function did not exist, but on the other hand there was one with the reverse effect called `isimutable`

I had added `Compat` as a depedency a while ago for the same reason, so that some functions in in `src/shortestpaths/johnson.jl` would work with older Julia versions. So this is not a new dependency.